metadata-editor/app/controllers/EditsController.scala

```diff
@@ -146,19 +144,32 @@ object EditsController extends Controller with ArgoHelpers with DynamoEdits {
   }
 
   def setMetadataFromUsageRights(id: String) = Authenticated.async { req =>
-    dynamo.jsonGet(id, "usageRights").flatMap { dynamoEntry =>
+    dynamo.get(id) flatMap { dynamoEntry =>
+      val edits = dynamoEntry.as[Edits]
+      val originalMetadata = edits.metadata
+      val metadataOpt = edits.usageRights.flatMap(usageRightsToMetadata)
 
-      val usageRights = (dynamoEntry \ "usageRights").asOpt[UsageRights]
-      val metadataOpt = usageRights.flatMap(metadataFromUsageRights)
       metadataOpt.map { metadata =>
-        dynamo.jsonPatch(id, "metadata", metadataAsMap(metadata))
+        // This merge is based on the fact that we know we will override byline and credit.
+        // TODO: Be more intelligent about this
+        val mergedMetadata = (metadata.credit, metadata.byline) match {
+          case (Some(_), Some(_)) =>
+            originalMetadata.copy(credit = metadata.credit, byline = metadata.byline)
+          case (Some(_), None) =>
+            originalMetadata.copy(credit = metadata.credit)
+          case (None, Some(_)) =>
+            originalMetadata.copy(byline = metadata.byline)
+          case _ => originalMetadata
+        }
+
+        dynamo.jsonAdd(id, "metadata", metadataAsMap(mergedMetadata))
           .map(publish(id))
-          .map(edits => respond(edits.metadata))
+          .map(edits => respond(edits.metadata, uri = Some(metadataUri(id))))
       }
       .getOrElse(Future.failed(EditsValidationError("no-matching-metadata-found", "Couldn't find any matching metadata")))
```

Comment:
Saying that -I wasn't handling the 404 - [I've added that now](https://github.com/guardian/grid/commit/ca24f94cbff568deb4f9e8049424767ef33bdc8e).